include/reactor-uc/network_channel.h

```diff
@@ -95,6 +107,19 @@ struct NetworkChannel {
   void (*free)(NetworkChannel *self);
 };
 
+struct SyncNetworkChannel {
+  NetworkChannel super;
+
+  /**
+   * @brief Polls for new data and calls the callback handler if a message is successfully decoded
+   */
+  void (*poll)(NetworkChannel *self);
+};
+
+struct AsyncNetworkChannel {
+  NetworkChannel super;
```

Comment:
Does it maybe make sense to move the async callback out of `NetworkChannel` and into `AsyncNetworkChannel` instead of only inheriting it?

Reply:
No, because it is also used by the PolledNetworkChannel